sensor/tests/resource/helper.go

```diff
@@ -150,6 +152,43 @@ func NewContextWithConfig(t *testing.T, config CentralConfig) (*TestContext, err
 	}, nil
 }
 
+// WithPermutation sets whether the test should run with permutations
+func WithPermutation() TestRunFunc {
+	return func(t *testRun) {
+		t.permutation = true
+	}
+}
+
+// WithResources sets the resources to be created by the test
+func WithResources(resources []K8sResourceInfo) TestRunFunc {
+	return func(t *testRun) {
+		t.resources = resources
+	}
+}
+
+// WithTestCase sets the TestCallback function to be run
+func WithTestCase(test TestCallback) TestRunFunc {
+	return func(t *testRun) {
+		t.testCase = test
+	}
+}
+
+// WithRetryCallback sets the RetryCallback function
+func WithRetryCallback(retryCallback RetryCallback) TestRunFunc {
+	return func(t *testRun) {
+		t.retryCallback = retryCallback
+	}
+}
+
+// NewRun runs a test case. Fails the test if the testRun cannot be created.
+func (c *TestContext) NewRun(options ...TestRunFunc) {
```

Comment:
I think `NewRun` implies the creation of an object (which is the case of `newTestRun`, for example). WDYT about renaming the public function to simply: `Run(...)`? `RunTest` or `Test` sound good also, because they are more imperative.

Reply:
I like `RunTest` or `Test` :)